pkg/stanza/fileconsumer/file.go

```diff
@@ -74,13 +62,23 @@ func (m *Manager) Start(persister operator.Persister) error {
 }
 
 func (m *Manager) closePreviousFiles() {
-	if len(m.knownFiles) > 4*m.movingAverageMatches {
-		m.knownFiles = m.knownFiles[m.movingAverageMatches:]
+	// m.previousPollFiles -> m.knownFiles[-1]
+
+	for r, _ := m.previousPollFiles.Pop(); r != nil; r, _ = m.previousPollFiles.Pop() {
+		m.knownFiles[len(m.knownFiles)-1].Add(r.Close())
 	}
-	for _, r := range m.previousPollFiles {
-		m.knownFiles = append(m.knownFiles, r.Close())
+}
+
+func (m *Manager) rotateFilesets() {
+	// rotate the filesets at end of every consume() call
+
+	// m.knownFiles[0] wil be removed
+	// m.knownFiles[1] -> m.knownFiles[0]
+	// m.knownFiles[2] -> m.knownFiles[1]
+	for i := 0; i < len(m.knownFiles)-1; i++ {
+		m.knownFiles[i] = m.knownFiles[i+1]
```

Comment:
Maybe a nit, but could we invert this? My feeling is that having index 0 be the most recent is more intuitive because we generally iterate up, and we want to work with most recent files first. It should be easy enough to shift the elements in place too. [example](https://go.dev/play/p/nR0k6PYWcC9)
```suggestion
	// shift the filesets at end of every consume() call
	// m.knownFiles[0] -> m.knownFiles[1] -> m.knownFiles[2]
	copy(m.knownFiles[1:], m.knownFiles)
	m.knownFiles[0] = fileset.New[*reader.Metadata](m.maxBatchFiles / 2)
```